### ISSUE TITLE
WPF GridView/DataGrid column width cannot be changed using only the keyboard

### ISSUE THREAD

**Issue by singhashish-wpf** [dotnet/wpf#5227]
Is this feature request related specifically to tooling in Visual Studio (e.g. XAML Designer, Code editing, etc...)? If yes, please file the request via the instructions here: https://docs.microsoft.com/visualstudio/ide/suggest-a-feature?view=vs-2019

Steps to reproduce:

See attachment WpfApp8DataGrid.zip. I have demo'd a GridView and DataGrid, and neither supports column resizing via Keyboard. This demo follows instructions from External Bug Process (sharepoint.com).

This is blocking NuGet Client on a Sev2 a11y bug: Bug 1359962: A11y_Visual Studio Benchmark-ASPNETCore_GRPC_Update package_Grpc.Asp.net core_Keyboard: Keyboard users are not provided with the... - Boards

Closed related issue: Bug 424448: WPF DataGrid column width cannot be changed using only the keyboard - Boards (visualstudio.com)

WinForms does support this. See: Bug 503194: Accessibility WinForms: The customer is unable to change the DataGridView column widths using only the keyboard - Boards (visualstudio.com)

Copied from DevDiv Work Item 1383796

**Comment by anjalisheel-wpf** [comment]
@[merriemcgaw](https://github.com/merriemcgaw) 

In this we are adding keyboard resize support for DataGrid and GridView. I have incorporated changes for DataGrid, and I have changes ready for GridView as well.

 

But for GridView, we are facing challenges while making this change as  we are not able to get the focus on GridHeadersColumns, if we will not be able to get the focus then how will we perform resize on the columns.

 

One way to get the focus is by adding properties for GridView at the time of creating it, but that adds an additional check to include those properties and then only it can get resized from the keyboard. This is the workaround.

 

We will have to make changes in the focusable model in order to get focus inside GridView, but that will be a major change in the sense that it will change the behaviour of “TabStops” in GridView.  So I wanted to check with you that what should be the approach for the same.

**Comment by vishalmsft** [comment]
cc: @Swati1700
duplicate: #5880
> 
> In this we are adding keyboard resize support for DataGrid and GridView. I have incorporated changes for DataGrid, and I have changes ready for GridView as well.
> 
> But for GridView, we are facing challenges while making this change as we are not able to get the focus on GridHeadersColumns, if we will not be able to get the focus then how will we perform resize on the columns.
> 
> One way to get the focus is by adding properties for GridView at the time of creating it, but that adds an additional check to include those properties and then only it can get resized from the keyboard. This is the workaround.
> 
> We will have to make changes in the focusable model in order to get focus inside GridView, but that will be a major change in the sense that it will change the behaviour of “TabStops” in GridView. So I wanted to check with you that what should be the approach for the same.

**Comment by dipeshmsft** [comment]
Closing since duplicate of #5880, and the code changes were already merged. See #5880 for more details
